cpp/src/arrow/compute/kernels/scalar_temporal_unary.cc

```diff
@@ -725,6 +725,27 @@ const Duration FloorTimePoint(const int64_t arg, const int64_t multiple,
   }
 }
 
+template <typename Duration, typename Unit, typename Localizer>
```

Comment:
nit: is there any point templating on Unit when it's apparently always `weeks` (and doesn't seem to make sense for any other unit)?

Reply:
Done.